### ISSUE TITLE
GsonConverterFactory (null HashMap key) issue with Castle

### ISSUE THREAD

**Issue by dlew** [castle/castle-android#37]
It looks like null keys are getting into Castle internal `HashMaps` somehow.

We've been seeing the following stack trace appear in Castle ever since upgrading to SDK v1.2.1 (we are on v1.2.2 right now, still seeing it):

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Object.hashCode()' on a null object reference
       at libcore.reflect.TypeVariableImpl.hashCode(TypeVariableImpl.java:47)
       at java.util.HashMap.hash(HashMap.java:338)
       at java.util.HashMap.containsKey(HashMap.java:595)
       at java.util.HashSet.contains(HashSet.java:203)
       at com.google.gson.internal.$Gson$Types.resolve(.java:346)
       at com.google.gson.internal.$Gson$Types.resolve(.java:381)
       at com.google.gson.internal.$Gson$Types.resolve(.java:337)
       at com.google.gson.internal.$Gson$Types.getSupertype(.java:283)
       at com.google.gson.internal.$Gson$Types.getCollectionElementType(.java:302)
       at com.google.gson.internal.bind.CollectionTypeAdapterFactory.create(CollectionTypeAdapterFactory.java:52)
       at com.google.gson.Gson.getAdapter(Gson.java:458)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.createBoundField(ReflectiveTypeAdapterFactory.java:117)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:166)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102)
       at com.google.gson.Gson.getAdapter(Gson.java:458)
       at retrofit2.converter.gson.GsonConverterFactory.requestBodyConverter(GsonConverterFactory.java:74)
       at retrofit2.Retrofit.nextRequestBodyConverter(Retrofit.java:315)
       at retrofit2.Retrofit.requestBodyConverter(Retrofit.java:293)
       at retrofit2.RequestFactory$Builder.parseParameterAnnotation(RequestFactory.java:778)
       at retrofit2.RequestFactory$Builder.parseParameter(RequestFactory.java:325)
       at retrofit2.RequestFactory$Builder.build(RequestFactory.java:206)
       at retrofit2.RequestFactory.parseAnnotations(RequestFactory.java:67)
       at retrofit2.ServiceMethod.parseAnnotations(ServiceMethod.java:26)
       at retrofit2.Retrofit.loadServiceMethod(Retrofit.java:202)
       at retrofit2.Retrofit$1.invoke(Retrofit.java:160)
       at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
       at $Proxy10.batch($Proxy10.java)
       at io.castle.android.queue.EventQueue.flush(EventQueue.java:112)
       at io.castle.android.Castle.flush(Castle.java:324)
       at io.castle.android.Castle.registerLifeCycleCallbacks(Castle.java:85)
       at io.castle.android.Castle.configure(Castle.java:128)
       at com.trello.app.TrelloApplication.initializeCastle(TrelloApplication.java:360)
       at com.trello.app.TrelloApplication.initialize(TrelloApplication.java:145)
       at com.trello.app.TrelloApplication.onCreate(TrelloApplication.java:78)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1183)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6479)
       at android.app.ActivityThread.access$1300(ActivityThread.java:220)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1860)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7397)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)
```

AFAIK, we did not see this in v1.2.0 or earlier (or at least, not nearly at such a rate).

It does not happen often (only a few people a day), and it seems to happen only once for each user it strikes.

-------

Our initialization is fairly simple, if that helps:

```
val config = CastleConfiguration.Builder()
    .debugLoggingEnabled(BuildConfig.DEBUG)
    .screenTrackingEnabled(true)
    .build()

Castle.configure(this, <castle_key>, config)
```

**Comment by sebastiansimson** [comment]
Thank you for reporting, we will look into this and get back to you.

**Comment by sebastiansimson** [comment]
After investigating this, it seems to be because of an error I made that reverted the retrofit/gson dependencies (in 1.2.1) that was upgraded in 1.2.0. This I think is resulting in some corruption of the data in the queue. 

I have created a PR with a fix that prevents the app from crashing and clears the queue which should prevent it from happening again. I will do some upgrade testing based on the versions and see if I can reproduce the issue before making a new release.

Thanks again for reporting!
